libs/reactive-forms/src/lib/types.ts

```diff
@@ -19,7 +19,7 @@ export type ValuesOf<T extends ControlsOf<any>> = {
   ? R
   : NonUndefined<T[K]> extends FormGroup<infer R>
   ? ValuesOf<R>
-  : NonUndefined<T[K]> extends FormArray<infer R> ? R[] : NonUndefined<T[K]>;
+  : NonUndefined<T[K]> extends FormArray<infer R, infer C> ? R[] : NonUndefined<T[K]>;
```

Comment:
infer C is redudant no?

Reply:
I know this is strange but this is the only way I found to have the type matching:
- `NonUndefined<T[K]> extends FormArray<infer R>` match `FormArray<R>` and return `R[]`
- `NonUndefined<T[K]> extends FormArray<infer R>` doesn't match `FormArray<R, FormControl<R>>` and return `NonUndefined<T[K]>`
- `NonUndefined<T[K]> extends FormArray<infer R, infer C>` match `FormArray<R, FormControl<R>>` and return `R[]` 

TBH I am not sure about this solution. I just want `ValuesOf` to work when specifying a `Control` type